### ISSUE TITLE
Error when using dropout in GRU

### ISSUE THREAD

**Issue by Trojahn** [Element-Research/rnn#415]
The following toy example works:

require 'rnn'

model = nn.Sequential()
model:add(nn.GRU(10, 5, 20))
model:add(nn.GRU(5, 1, 20))
model:add(nn.Sigmoid())
model = nn.Sequencer(model)

t = torch.Tensor(20,10)
model:forward(t)
print(model.output)

But using the available dropout parameter don't:

require 'rnn'

model = nn.Sequential()
**model:add(nn.GRU(10, 5, 20, 0.25))**
model:add(nn.GRU(5, 1, 20))
model:add(nn.Sigmoid())
model = nn.Sequencer(model)

t = torch.Tensor(20,10)
model:forward(t)
print(model.output)

The error itself...
In 1 module of nn.Sequential:
In 1 module of nn.Sequential:
In 2 module of nn.ConcatTable:
In 1 module of nn.Sequential:
In 1 module of nn.ParallelTable:
In 3 module of nn.Sequential:
bad argument #2 to '?' (out of bounds at /tmp/luarocks_torch-scm-1-9229/torch7/lib/TH/generic/THStorage.c:202)
...

**Comment by tastyminerals** [comment]
I am getting the same issue but using the already trained checkpoint.

**Comment by tastyminerals** [comment]
If you disable lazy `nn.Dropout` in GRU gates the error does not appear anymore. So, you can use `nn.Dropout` module after GRU without crashes, this does not help to improve the training as much as when with `nn.Dropout` used in GRU gates.

**Comment by tastyminerals** [comment]
Check your input to the model, it is probably one dimensional but should be two dimensional.
